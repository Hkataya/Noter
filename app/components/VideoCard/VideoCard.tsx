import React from 'react';
import styled from 'styled-components';
import { VideoType } from '../../reducers/entities/types';
import MenuButton from '../Button/MenuButton';

const Wrapper = styled.div.attrs({
  className:
    'w-full flex border-r border-b border-l border-gray-400 border-l-0 border-t border-gray-400 bg-white rounded-b rounded-b-none rounded-r mr-5 mb-5'
})``;

const First = styled.div.attrs({
  className: 'w-32 h-auto bg-cover bg-center rounded-t rounded-l'
})``;

const Second = styled.div.attrs({
  className: 'p-3 flex-auto flex flex-col justify-between'
})``;

const Third = styled.div.attrs({
  className: ' p-3 flex justify-center'
})``;

const Fourth = styled.div.attrs({
  className: ' pr-2 pl-3'
})``;

type Props = VideoType & {
  onRemoveClick: () => void;
  directToMediaPage: () => void;
  onToggleClick: () => void;
  onVideoCardClick: () => void;
  onUpdateVideoClick: () => void;
};

const VideoCard = (props: Props) => {
  const {
    title,
    onRemoveClick,
    directToMediaPage,
    onToggleClick,
    onVideoCardClick,
    watched,
    onUpdateVideoClick
  } = props;

  const checkColor: string = watched ? 'text-green-500' : 'text-gray-500';
  const items = [
    {
      label: 'remove',
      action: onRemoveClick
    },
    {
      label: 'update',
      action: onUpdateVideoClick
    }
  ];
  return (
    <Wrapper onClick={onVideoCardClick}>
      <Second>
        <div className="mb-8">
          <div className="text-gray-900 font-bold text-lg mb-2">
            <button
              onClick={e => {
                e.stopPropagation();
                directToMediaPage();
              }}
              type="button"
            >
              {title}
            </button>
          </div>
        </div>
      </Second>
      <Third>
        <button
          data-testid="watched-toggle"
          className="focus:outline-none"
          type="button"
          onClick={e => {
            e.stopPropagation();
            onToggleClick();
          }}
        >
          <i className={`fas fa-check fa-2x ${checkColor} `} />
        </button>
      </Third>
      <Fourth>
        <MenuButton items={items} />
      </Fourth>
    </Wrapper>
  );
};

export default VideoCard;
