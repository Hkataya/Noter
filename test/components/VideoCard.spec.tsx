import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import VideoCard from '../../app/components/VideoCard/VideoCard';

test('loads and displays video card', async () => {
  const { asFragment } = render(
    <VideoCard
      id="123"
      title="test"
      section="456"
      url="testurl"
      watched={false}
      onRemoveClick={() => {}}
      onToggleClick={() => {}}
      onVideoCardClick={() => {}}
      directToMediaPage={() => {}}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

test('clicks watched toggle', async () => {
  const mockFn = jest.fn();

  render(
    <VideoCard
      id="123"
      title="test"
      section="456"
      url="testurl"
      watched
      onRemoveClick={() => {}}
      onToggleClick={mockFn}
      directToMediaPage={() => {}}
      onVideoCardClick={() => {}}
    />
  );
  fireEvent.click(screen.getByTestId('watched-toggle'));
  expect(mockFn).toHaveBeenCalled();
});
