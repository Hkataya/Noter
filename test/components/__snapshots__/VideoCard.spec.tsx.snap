// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays video card 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb ccMvAG w-full flex border-r border-b border-l border-gray-400 border-l-0 border-t border-gray-400 bg-white rounded-b rounded-b-none rounded-r mr-5 mb-5"
  >
    <div
      class="sc-AxhUy dBRLnL w-32 h-auto bg-cover bg-center rounded-t rounded-l"
    />
    <div
      class="sc-AxgMl fFPQpl p-3 flex-auto flex flex-col justify-between"
    >
      <div
        class="mb-8"
      >
        <div
          class="text-gray-900 font-bold text-lg mb-2"
        >
          <button
            type="button"
          >
            test
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-AxheI fniENP  p-3 flex justify-center"
    >
      <button
        class="focus:outline-none"
        data-testid="watched-toggle"
        type="button"
      >
        <i
          class="fas fa-check fa-2x text-gray-500 "
        />
      </button>
    </div>
    <div
      class="sc-Axmtr hBVTto  pr-2 pl-3"
    >
      <div
        class="relative inline-block"
      >
        <div>
          <button
            class="sc-AxjAm StDqN inline-flex justify-center w-full rounded-md  text-sm leading-5 font-medium text-gray-700 hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue active:bg-gray-50 active:text-gray-800 transition ease-in-out duration-150"
            role="button"
            type="button"
          >
            <i
              class="fas fa-ellipsis-v"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
